codegen/grid.py

```diff
@@ -717,6 +720,13 @@ def set_variable(self, var, value=0, type='int', constant=False):
             var = var.name
         self.defined_variable[var] = Variable(var, value, type, constant)
 
+    def set_page_size(self, page_size):
```

Comment:
In that case stick to the original posix_memalign implementation that you had.

I am not happy for this to be merged.

On 21 Aug 2015, at 13:33, tj-sun <notifications@github.com<mailto:notifications@github.com>> wrote:


In codegen/grid.py<https://github.com/opesci/propagator/pull/25#discussion_r37629606>:

> @@ -717,6 +720,13 @@ def set_variable(self, var, value=0, type='int', constant=False):
>              var = var.name
>          self.defined_variable[var] = Variable(var, value, type, constant)
>
> +    def set_page_size(self, page_size):


I meant that I'm not sure how to allocate on heap using attribute, apart from using global variables (which I never used), the link seems to suggest using posix. I will add the generated code to the repo (src/tests/test3d.cpp), it looks something like this

float *_Txx_vec;
#ifdef _MSC_VER
_Txx_vec = (float*) _aligned_malloc(2315250*sizeof(float), 4096);
#else
posix_memalign((void **)(&_Txx_vec), 4096, 2315250*sizeof(float));
#endif
float (*Txx)[dim1][dim2][dim3]= (float (*)[dim1][dim2][dim3]) _Txx_vec;


—
Reply to this email directly or view it on GitHub<https://github.com/opesci/propagator/pull/25/files#r37629606>.